Fix tests failing in the browser and other test improvements.

jansiegel opened the pull request:
### Context
Some tests were only failing in the browser, this change should fix them.

**Note: when running tests in the browser, it's important not to do anything and keep the tab focused, as, for example, changing the focus to another window may mess up the `focus` event recognition, and simply moving the cursor might close the comments pop-up, resulting in a failed test.**

Additionally, I modified the `ContextMenu` plugin and `DateEditor` to prevent leaving behind containers after destroying the Handsontable instances.

### Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)

budnix commented on src/3rdparty/walkontable/test/spec/settings/stretchH.spec.js:
```diff
@@ -69,12 +69,15 @@ describe('stretchH option', () => {
     expect(currentTableWidth).toBeAroundValue($table[0].clientWidth);
     expect(currentTableWidth).toBeGreaterThan(initialTableWidth);
+
+    done();
```
`done()` call is not necessary when `async/await` is used.

jansiegel replied:
Without `done` the test would timeout before the async part was finished:
![image](https://user-images.githubusercontent.com/7397134/60669365-73543380-9e6e-11e9-84e7-287c90ecd8bf.png)

After updating Jasmine to the latest version this seem to not happen anymore, so I removed it.

budnix commented on src/editors/dateEditor.js:
```diff
@@ -74,6 +74,7 @@ class DateEditor extends TextEditor {
    */
   destroyElements() {
     this.$datePicker.destroy();
+    this.datePicker.remove();
```
`.remove()` method is not supported by IE (https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove)

budnix commented on src/plugins/contextMenu/menu.js:
```diff
@@ -265,6 +265,7 @@ class Menu {
     this.close();
     this.parentMenu = null;
     this.eventManager.destroy();
+    this.container.remove();
```
`.remove()` method is not supported by IE (https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove). Additionally, can I ask you to add/update test which will test if that element was removed?

jansiegel commented:
Updates after the CR:
- Updated Jasmine to the latest version (`2.5.2` -> `^3.4.0`)
- More leftover DOM elements removed
- Fixed Firefox and retina problems
- Temporarily disabled some tests - to be re-enabled after fixing  #6083 and #6085 (and possible remove the ones related with table captions)

If I'm not mistaken, these changes seem to have fixed (or at least worked around) the Chrome memory leak problem.

budnix commented on src/validators/dateValidator.js:
```diff
@@ -18,9 +18,9 @@ export default function dateValidator(value, callback) {
-  // is it in the specified format
+
   let isValidFormat = moment(valueToValidate, this.dateFormat || dateEditor.defaultDateFormat, true).isValid();
+  let isValidDate = moment(new Date(valueToValidate)).isValid() || isValidFormat;
```
A good catch 👍
